lib/puppet/functions/stdlib/parsehocon.rb

```diff
@@ -17,16 +21,18 @@
   end
 
   def parsehocon(hocon_string, default = :no_default_provided)
-    require 'hocon/config_factory'
-
-    begin
+    if File.exist? hocon_string
```

Comment:
I'm not a fan of this, because this can also run on the server. It becomes unpredictable.

Isn't there a way to read a file in puppet? I know [`file()`](https://github.com/puppetlabs/puppet/blob/main/lib/puppet/parser/functions/file.rb) can be used in some cases. Its docs mention it can also read absolute files, so I'd expect this to work:
```puppet
stdlib::parsehocon(file('/path/to/file'))
```

Reply:
yes the idea was to read a file from a PE installation on the puppetserver. Your solution should work. I think I will propose this function to the hocon module in case that's still maintained.